pkg/diagnostics/v2/client.go

```diff
@@ -0,0 +1,53 @@
+package v2
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+// Bundle represents a bundle object received from the diagnostics API
+type Bundle struct {
+	ID      string    `json:"id,omitempty"`
+	Size    int64     `json:"size,omitempty"` // length in bytes for regular files; 0 when Canceled or Deleted
+	Status  Status    `json:"status"`
+	Started time.Time `json:"started_at,omitempty"`
+	Stopped time.Time `json:"stopped_at,omitempty"`
+	Errors  []string  `json:"errors,omitempty"`
+}
+
+// Client is a REST API wrapper around the new Diagnostics API.
+type Client struct {
+	http *httpclient.Client
+}
+
+// NewClient creates a new Client.
+func NewClient(baseClient *httpclient.Client) *Client {
+	return &Client{
+		http: baseClient,
+	}
+}
+
+// List gets a list of all cluster bundles.
+func (c *Client) List() ([]Bundle, error) {
+	resp, err := c.http.Get("/system/health/v1/diagnostics")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var bundles []Bundle
+		err = json.NewDecoder(resp.Body).Decode(&bundles)
+		if err != nil {
+			return nil, err
+		}
+		return bundles, err
+	default:
+		return nil, fmt.Errorf("received unexpected status code %d", resp.StatusCode)
```

Comment:
Is there a list of known HTTP error codes that the diagnostics API might return and can be tied to more informative error messages?

At the very least, we should handle 401 and 403 status codes, see https://github.com/dcos/dcos-core-cli/blob/1.14-patch.x/pkg/diagnostics/client.go for example, with the httpResponseToError helper.

The generic 401 and 403 messages are handled here : https://github.com/dcos/dcos-core-cli/blob/1.14-patch.x/cmd/dcos/main.go

Reply:
In other endpoints yes but not for `List`, I've added the use of the `httpResponseToError` helper though.